# 📊 마케팅 전략 A와 B의 매출 증가율 비교  

한 기업은 기존 전략 B에 비해 신규 마케팅 전략 A가 매출 증가율을 더 높일 수 있는지 확인하고자 합니다.  

각 전략을 적용한 고객의 한 달 후 매출 증가율 데이터는 다음과 같습니다.  

In [2]:
import numpy as np
from scipy.stats import levene, ttest_ind

# 데이터
np.random.seed(42)
strategy_a = np.random.normal(loc=15, scale=4, size=40)
strategy_b = np.random.normal(loc=12, scale=7, size=35)

## 🔹 문제 1. 가설 설정

두 전략의 평균 매출 증가율을 비교하기 위한 **귀무가설(H₀)**과 **대립가설(H₁)**을 작성하시오.  
(양측 검정, 단측 검정 중 적절한 것 선택)

- 귀무가설(H₀) : 마케팅 전략 A의 평균 매출 증가율은 전략 B보다 크지 않다.  
    (즉, 전략 A가 전략 B보다 더 효과적이라고 할 수 없다.)  

- 대립가설(H₁) :  마케팅 전략 A의 평균 매출 증가율은 전략 B보다 크다.  
    (즉, 전략 A가 전략 B보다 매출 증가에 더 효과적이다.)  

- 단측 검정 사용  

## 🔹 문제 2. 편차(분산) 확인

전략 A와 B의 표본 평균과 표준편차를 계산하고,  
두 전략의 표준편차를 비교하여 어느 전략의 변동성이 더 큰지 간단히 서술하시오.

In [3]:
# 평균과 표준편차
mean_a, std_a = np.mean(strategy_a), np.std(strategy_a, ddof=1)
mean_b, std_b = np.mean(strategy_b), np.std(strategy_b, ddof=1)

print(f"A: 평균={mean_a:.4f}, 표준편차={std_a:.4f}")
print(f"B: 평균={mean_b:.4f}, 표준편차={std_b:.4f}")

A: 평균=14.1255, 표준편차=3.8112
B: 평균=12.0248, 표준편차=6.7487


변동성이 더 큰 전략은?

➡️ 전략 A의 평균 매출 증가율이 전략 B보다 높고,  
➡️ 전략 B는 표준편차가 더 커서 성과 변동성이 더 크다. 

## 🔹 문제 3. 평균 비교 검정 (t-검정)

3-1. 두 전략의 평균 매출 증가율을 비교하기 위해 이표본 t-검정을 수행하여 t-value과 p-value를 출력하시오.  

3-2. 유의수준 0.05에서 전략 A가 전략 B보다 매출 증가율이 높다고 볼 수 있는지 결론을 내리시오.  

In [4]:
# 3-1. 이표본 t-검정으로 t-value과 p-value 구하기

t_stat, p_value = ttest_ind(
    strategy_a,
    strategy_b,
    alternative='greater'  # A > B 단측 검정
)

print(f"t-valueㅍ: {t_stat:.4f}")
print(f"p-value: {p_value:.4f}")

t-valueㅍ: 1.6861
p-value: 0.0480


유의수준 0.05에서 전략 A가 전략 B보다 매출 증가율이 높다고 볼 수 있는가?

➡️ p-value = 0.0480 < 0.05  
➡️ 유의수준 0.05에서 마케팅 전략 A가 전략 B보다 매출 증가율이 더 높다고 볼 수 있으므로, 귀무가설을 기각한다.  
    (마케팅 전략 A의 평균 매출 증가율은 전략 B보다 크다는 대립가설 채택)